src/main/java/net/mcreator/element/converter/v2023_4/StructureToFeatureConverter.java

```diff
@@ -0,0 +1,278 @@
+			int minCountPerChunk = definition.get("minCountPerChunk").getAsInt();
+			int maxCountPerChunk = definition.get("maxCountPerChunk").getAsInt();
+			List<String> restrictionBlocks = definition.getAsJsonArray("restrictionBlocks").asList().stream()
+					.map(e -> e.getAsJsonObject().get("value").getAsString()).toList();
+			String placementXML = getPlacementXML(surfaceDetectionType, spawnProbability, minCountPerChunk,
+					maxCountPerChunk, spawnLocation, restrictionBlocks);
+
+			feature.featurexml = """
+					<xml><block type="feature_container" deletable="false" x="40" y="40">
+					<value name="feature">%s</value>
+					<next>%s</next></block></xml>
+					""".formatted(patchXML, placementXML);
+
+			return feature;
+		} catch (Exception e) {
+			LOG.warn("Could not convert old structure to a feature", e);
+		}
+
+		return null;
+	}
+
+	@Override public int getVersionConvertingTo() {
+		return 52;
+	}
+
+	// Returns the XML for the random patch block attached to the feature starting point
+	private static String getFeatureXML(String structure, int ox, int oy, int oz, boolean randomlyRotateStructure,
+			String ignoreBlocks) {
+		String ignoredBlocks;
+		if (ignoreBlocks.equals("STRUCTURE_BLOCK")) {
+			ignoredBlocks = "<mutation inputs=\"1\"></mutation><field name=\"block0\">Blocks.STRUCTURE_BLOCK</field>";
+		} else if (ignoreBlocks.equals("AIR_AND_STRUCTURE_BLOCK")) {
+			ignoredBlocks = "<mutation inputs=\"2\"></mutation><field name=\"block0\">Blocks.STRUCTURE_BLOCK</field><field name=\"block1\">Blocks.AIR</field>";
+		} else {
+			ignoredBlocks = "<mutation inputs=\"1\"></mutation><field name=\"block0\">Blocks.AIR</field>";
+		}
+		return """
+				<block type="feature_custom_structure">
+					<field name="structure">%s</field>
+					<field name="x">%d</field>
+					<field name="y">%d</field>
+					<field name="z">%d</field>
+					<field name="random_rotation">%s</field>
+					<field name="random_mirror">%s</field>
+					<value name="ignored_blocks">
+						<block type="block_holderset_list">%s</block>
+					</value>
+				</block>
+				""".formatted(structure, ox, oy, oz, randomlyRotateStructure ? "TRUE" : "FALSE",
+				randomlyRotateStructure ? "TRUE" : "FALSE", ignoredBlocks);
+	}
+
+	// Returns the XML for the feature placement
+	private static String getPlacementXML(String surfaceDetectionTypeRaw, int spawnProbability, int minCountPerChunk,
+			int maxCountPerChunk, String spawnLocation, List<String> restrictionBlocks) {
+		StringBuilder xml = new StringBuilder();
+		int blocksToClose = 0;
+
+		if (minCountPerChunk != 1 || maxCountPerChunk != 1) {
+			xml.append("""
+					<block type="placement_count">
+						<value name="count">
+							<block type="int_provider_uniform">
+								<field name="min">%d</field>
+								<field name="max">%d</field>
+							</block>
+						</value>
+						<next>""".formatted(minCountPerChunk, maxCountPerChunk));
+			blocksToClose++;
+		}
+
+		xml.append("""

```

Comment:
The rarity placement should be moved before the count placement, to preserve previous behavior